wagtail/contrib/modeladmin/options.py

```diff
@@ -28,33 +28,37 @@ class WagtailRegisterable:
     exclude_from_explorer = False
 
     def register_with_wagtail(self):
+        self.register_permissions_with_wagtail()
+        self.register_admin_urls_with_wagtail()
+        self.register_admin_menu_item_with_wagtail()
+        self.register_explorer_page_queryset_with_wagtail()
```

Comment:
Descriptive naming is good, but a `ModelAdmin` class only registers itself with Wagtail, so I think we can probably lose the `_with_wagtail` from these new method names.